activesupport/lib/active_support/message_encryptor.rb

```diff
@@ -84,7 +84,8 @@ def initialize(secret, *signature_key_or_options)
 
     # Encrypt and sign a message. We need to sign the message in order to avoid
     # padding attacks. Reference: http://www.limited-entropy.com/padding-oracle-attacks.
-    def encrypt_and_sign(value)
+    def encrypt_and_sign(value, meta_data = {})
+      value = { value: value, meta_data: meta_data } unless meta_data.empty?
```

Comment:
When I said metadata I didn't mean users could any metadata, but a name to convey what the purpose and expiry are at a higher level 😊

For this to work we also need to add support for purpose when decrypting and verifying, so the original purpose within the message can be compared against (the expiry has the info within it). See the SignedGlobalID class for the API we need.